src/java/htsjdk/samtools/SAMUtils.java

```diff
@@ -970,4 +971,77 @@ public static String getCanonicalRecordName(final SAMRecord record) {
         else name = name + ":" + record.getReadName();
         return name;
     }
+
+    /**
+     * Returns the number of bases that need to be clipped due to overlapping pairs.  If the record is not paired,
+     * or the given record's start position is greater than its mate's start position, zero is automatically returned.
+     *
+     * @param rec
+     * @return the number of bases at the end of the read that need to be clipped such that there would be no overlapping bases with its mate.
+     */
+    public static int getNumOverlappingBasesToClip(final SAMRecord rec) {
+        // NB: ignores how to handle supplemental records when present for both ends by just using the mate information in the record.
+
+        if (!rec.getReadPairedFlag() || rec.getReadUnmappedFlag() || rec.getMateUnmappedFlag()) return 0;
+
+        // Only clip records that are left-most in genomic order and overlapping.
+        if (rec.getMateAlignmentStart() < rec.getAlignmentStart()) return 0; // right-most, so ignore.
+
+        final int numBasesToClip = rec.getAlignmentEnd() - rec.getMateAlignmentStart()  + 1;
```

Comment:
hmmm. I think that there's a mistake here. 

The overlap is based on reference bases, but the clipping is done on read bases. So if there's an insertion or deletion in the overlap region, I suspect that the calculation will be wrong.

Could you add a few more test cases that have insertions and/or deletions in the overlap region?